did/ion/model.go

```diff
@@ -101,18 +119,85 @@ type UpdateSignedDataObject struct {
 }
 
 type Delta struct {
-	Patches          []any  `json:"patches,omitempty"` //revive:disable-line
-	UpdateCommitment string `json:"updateCommitment,omitempty"`
+	Patches          []Patch `json:"patches,omitempty"` //revive:disable-line
+	UpdateCommitment string  `json:"updateCommitment,omitempty"`
+}
+
+func (d *Delta) UnmarshalJSON(data []byte) error {
+	var deltaMap map[string]any
+	if err := json.Unmarshal(data, &deltaMap); err != nil {
+		return errors.Wrap(err, "unmarshalling patch to generic map")
+	}
+	updateCommitment, ok := deltaMap["updateCommitment"].(string)
+	if !ok {
+		return fmt.Errorf("no updateCommitment found in delta")
+	}
+	d.UpdateCommitment = updateCommitment
+	allPatches, ok := deltaMap["patches"].([]any)
+	if !ok {
+		return fmt.Errorf("no patches found in delta")
+	}
+	var patches []Patch
+	for _, patch := range allPatches {
+		currPatch, ok := patch.(map[string]any)
+		if !ok {
+			return fmt.Errorf("patch is not a map")
+		}
+		action, ok := currPatch["action"]
+		if !ok {
+			return fmt.Errorf("patch has no action")
+		}
+		currPatchBytes, err := json.Marshal(currPatch)
```

Comment:
I think there is a way to use json.RawMessage to avoid the whole unmarshal -> marshal -> unmarshal round tripping, but I've never done it. If you're feeling brave :D

Reply:
hm you're right but I am not feeling brave